tools/docs_parser.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import json
+import re
+from collections import defaultdict
+import yaml
+
+
+
+def parse_route_line(line: str):
+    verb = re.match(r"routes:([^\s]+)\(", line).groups()[0]
+    url = re.findall(r"\@/:.*\"", line)[0][1:-1]
+    params = re.findall(r"function\((.*)", line)[0][:-1].split(", ")[1:]
+
+    parameters = []
+    for param_type, param in zip(re.finditer(r"\:[a-z]+", url), params):
+        param_type = param_type.group(0)
+        url = url.replace(param_type, f"{{{param}}}", 1)
+        parameters.append(
+            {
+                "name": param,
+                "in": "path",
+                "required": True,
+                "schema": {"type": param_type[1:]},
+            }
+        )
+
+    return verb, url, parameters
+
+
+def parse_route_comment(comment):
+    try:
+        yml = yaml.safe_load(comment)
+    except yaml.scanner.ScannerError:
+        yml = {}
+
+    request_body = {}
+    if "bodyContentType" in yml:
+        request_body = {
+            yml["bodyContentType"]: {
+                "schema": {"type": "object", "example": "your doc text"}
+            }
+        }
+
+    responses = {
+        code: content or {}
+        for code, content in yml.get("responses", {}).items()
+    }
+
+    description = yml.get("description", ""),
+
+    return description, request_body, responses
+
+def parse(input: str, output: str):
+    """
+    Get swagger inline: https://github.com/readmeio/swagger-inline
```

Comment:
Hmm.. can't we just use off-the-shelf swagger-inline then?

Reply:
If it doesn't support Lua comments, then we could just adjust the script to extract them and replae with JS-style comments. The simpler it is on our side, the better.

Reply:
I couldnt find anything with lua sadly, and this one is so convoluted that Ive endup going with python.
Also we could just go with generating manually, actually seen the comments, we could keep it in the same file kinda how it is now, and only extract the comments as big yaml file

Reply:
I was thinking about: 1) prefix the comments with some special characters (e.g. four dashes); 2) `grep` them out forming a new file; 3) pass to a Swagger generator.